plugins/catalog-react/src/apis/EntityPresentationApi/EntityPresentationApi.ts

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2023 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Entity } from '@backstage/catalog-model';
+import {
+  ApiRef,
+  IconComponent,
+  createApiRef,
+} from '@backstage/core-plugin-api';
+import { Observable } from '@backstage/types';
+
+/**
+ * An API that handles how to represent entities in the interface.
+ *
+ * @public
+ */
+export const entityPresentationApiRef: ApiRef<EntityPresentationApi> =
+  createApiRef({
+    id: 'plugin.catalog.entity-presentation',
+  });
+
+/**
+ * The visual presentation of an entity reference at some point in time.
+ *
+ * @public
+ */
+export interface EntityRefPresentationSnapshot {
+  /**
+   * The ref to the entity that this snapshot represents.
+   *
+   * @remarks
+   *
+   * Note that when the input data was broken or had missing vital pieces of
+   * information, this string may contain placeholders such as "unknown". You
+   * can therefore not necessarily assume that the ref is completely valid and
+   * usable for example for forming a clickable link to the entity.
+   */
+  entityRef: string;
+  /**
+   * The entity that this snapshot represents.
+   *
+   * @remarks
+   *
+   * Note that this may be a partial entity, i.e. it may not contain all of the
+   * fields that the kind at hand normally contains, since the input to the
+   * representation renderer might have been a partial entity. So be careful
+   * when accessing fields on this.
+   */
+  entity?: Entity | undefined;
```

Comment:
Probably remove this for now? Should be unnecessary. Then again if we remove filters then this is more safe to keep, although on top of that one might replace the entity presentation API implementation with one that doesn't fetch complete entities, at which point we're back to this being quite risky.

The documentation here that states that the entity might be partial will not be enough to keep people from writing broken code consuming this field 😅

Reply:
fixed in [54cbdc1](https://github.com/backstage/backstage/pull/17474/commits/430f2931b30efb373ea341b9c93f2f02054cbdc1)